ddtrace/contrib/pytest/plugin.py

```diff
@@ -116,29 +192,48 @@ def pytest_runtest_protocol(item, nextitem):
         yield
         return
 
-    with _CIVisibility._instance.tracer.trace(
+    if isinstance(item.parent.parent, pytest.Package):
+        if _extract_span(item.parent.parent) is None:
+            _start_test_module_span(item)
+
+    if _extract_span(item.parent) is None:
+        _start_test_suite_span(item)
+    module_span = _extract_span(item.parent)
+
+    with _CIVisibility._instance.tracer.start_span(
         ddtrace.config.pytest.operation_name,
+        child_of=module_span,
```

Comment:
⚠️ We want to keep tests as independent traces, so that the test traces can be navigated without the rest of the session. Does this `child_of` mean that the `trace_id` and `parent_id` of the test will be that of the module?

Reply:
Hmm. My understanding was that test spans part of the same session are all part of the same trace, and all test event spans (in the same file) should be child spans of the test suite (file) span, and all test suite spans (in the same package) should be child spans of the test module span, and etc. Is that not correct?

Reply:
this is correct 👍 

> The test event spans are not children of any spans in the test session trace at tracing time, but the backend will handle that test suite visibility logic.

that's correct. If `test_session_id` and the rest of the ids are present in the test event, the backend will handle the logic.